bazel/thirdparty/avro.bzl

```diff
@@ -0,0 +1,38 @@
+"""
+This module contains functions for working with Avro.
+"""
+
+load("//bazel:build.bzl", "redpanda_cc_library")
+
+def avro_cc_library(name, schema, namespace, include_prefix = None, visibility = None):
+    """
+    Generate Avro library.
+
+    Args:
+      name: name of the library
+      schema: the schema file
+      namespace: namespace for generated code
+      include_prefix: header include prefix
+      visibility: library visibility
+    """
+    hh_out = name + ".h"
+
+    native.genrule(
+        name = name + "_genrule",
+        srcs = [schema],
+        outs = [hh_out],
+        cmd = "$(location @avro//:avrogen) -i $< -o $@ -n " + namespace,
```

Comment:
Oh BTW it's recommended to use `execpath` now because `location` does different things depending on how it's used: https://bazel.build/reference/be/make-variables#predefined_label_variables

Reply:
Thanks. There are several spots that need updated i'll do them all in a separate PR.

Reply:
SG